src/item.cpp

```diff
@@ -4842,36 +4823,59 @@ bool item::fill_with( item &liquid, std::string &err, bool allow_bucket )
     return true;
 }
 
-bool item::use_charges(const itype_id &it, long &quantity, std::list<item> &used)
+bool item::use_charges( const std::string& what, long& qty, std::list<item>& used, const tripoint& pos )
```

Comment:
consistency with `player::use_charges` which can be passed pseudo-id's, eg `fire`, `apparatus`